src/Raven.Client/Documents/Session/DocumentSession.Patch.cs

```diff
@@ -35,37 +82,36 @@ public partial class DocumentSession
         public void Increment<T, U>(string key, Expression<Func<T, U>> path, U valToAdd)
         {
             var pathScript = path.CompileToJavascript();
-            var script = $"this.{pathScript} += val;";
 
-            _documentStore.Operations.Send(new PatchOperation(key, null, patch: new PatchRequest
+            Advanced.Defer();
```

Comment:
Signature is misleading, maybe change to two overloads:
1. Defer(ICommandData command, params ICommandData[] commands)
2. Defer(ICommandData[] commands) ?